src/main/java/org/gridsuite/modification/server/entities/equipment/deletion/ByFilterDeletionEntity.java

```diff
@@ -0,0 +1,84 @@
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.gridsuite.modification.server.dto.ByFilterDeletionInfos;
+import org.gridsuite.modification.server.dto.FilterInfos;
+import org.gridsuite.modification.server.dto.ModificationInfos;
+import org.gridsuite.modification.server.entities.ModificationEntity;
+import org.gridsuite.modification.server.entities.equipment.modification.VariationFilterEntity;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author Antoine Bouhours <antoine.bouhours at rte-france.com>
+ */
+@NoArgsConstructor
+@Getter
+@Entity
+@Table(name = "byFilterDeletion")
+@PrimaryKeyJoinColumn(foreignKey = @ForeignKey(name = "by_filter_deletion_id_fk_constraint"))
+public class ByFilterDeletionEntity extends ModificationEntity {
+    @Column(name = "equipmentType")
+    private String equipmentType;
+
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "byFilterDeletionFilters",
+            joinColumns = @JoinColumn(name = "id"), foreignKey = @ForeignKey(name = "by_filter_deletion_id_fk"),
+            inverseJoinColumns = @JoinColumn(name = "filterId"), inverseForeignKey = @ForeignKey(name = "variation_filter_id_fk"))
+    private List<VariationFilterEntity> filters;
+
+    public ByFilterDeletionEntity(ByFilterDeletionInfos byFilterDeletionInfos) {
+        super(byFilterDeletionInfos);
+        assignAttributes(byFilterDeletionInfos);
+    }
+
+    @Override
+    public void update(@NonNull ModificationInfos modificationInfos) {
+        super.update(modificationInfos);
+        assignAttributes((ByFilterDeletionInfos) modificationInfos);
+    }
+
+    private void assignAttributes(ByFilterDeletionInfos byFilterDeletionInfos) {
+        this.equipmentType = byFilterDeletionInfos.getEquipmentType();
+        if (filters == null) {
+            this.filters = byFilterDeletionInfos.getFilters().stream().map(FilterInfos::toEntity).collect(Collectors.toList());
+        } else {
+            filters.clear();
+            filters.addAll(byFilterDeletionInfos.getFilters().stream().map(FilterInfos::toEntity).collect(Collectors.toList()));
+        }
+    }
+
+    @Override
+    public ByFilterDeletionInfos toModificationInfos() {
+        return ByFilterDeletionInfos
+                .builder()
+                .uuid(getId())
+                .date(getDate())
+                .stashed(getStashed())
+                .filters(this.getFilters().stream()
+                        .map(filter -> new FilterInfos(filter.getFilterId(), filter.getName()))
+                        .collect(Collectors.toList()))
+                .equipmentType(getEquipmentType()).build();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        return super.equals(o);
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode();
+    }
```

Comment:
I had a Sonarlint issue because of the override of the `equals` and `hashcode` methods in the ModificationEntity class. It was removed in https://github.com/gridsuite/network-modification-server/pull/399/files so I will get rid of it.